### ISSUE TITLE
How to combine bus stop description as a CompletableFuture?

### ISSUE THREAD

**Issue by lewistham9x** [nus-cs2030/2122-s1#258]
# Summary

I am unsure how to combine the describption function of the BusRoutes file as a CompletableFuture.

## Description:

How do you chain multiuple completable futures together?

**Comment by KeithPJX** [comment]
I had a class variable `final CompletableFuture<Map<BusService,CompletableFuture<Set<BusStop>>>> services;`

I then did a `.thenComposeAsync` of `this.services` and kept the for loop header they gave us in `description()`. Since the map now has values of type `CompletableFuture<Set<BusStop>>`, I just did a `.thenCombine`  with `CompletableFuture<String> desc = CompletableFuture.completedFuture(result);`.

**Comment by lewistham9x** [comment]
Thanks Keith!

**Comment by mewrion** [comment]
Somewhat similar to Keith. I retained most of the original code by wrapping the` Set<BusStop>` in a CF, but did not wrap map. Instead, I wrap the string that I need to return, and use `thenCombine `and iterative calls to `describeServices` and concatenated with lambda functions

**Comment by jackhocheukhei** [comment]
Yep similar to them, for every BusService, wrap the strings using thenCombine and concatenate the results with lambda functions for every iterative call.
